packages/hebao_v1/contracts/modules/security/GuardianUtils.sol

```diff
@@ -61,12 +43,12 @@ library GuardianUtils
         view
     {
         // Calculate total group sizes
-        SecurityStore.Guardian[] memory allGuardians = securityStore.guardians(wallet);
-        GuardianGroupSizes memory total = countGuardians(allGuardians);
+        Data.Guardian[] memory allGuardians = securityStore.guardians(wallet);
+        uint[16] memory total = countGuardians(allGuardians);
 
         // Calculate how many signers are in each group
         bool walletOwnerSigned = false;
-        SecurityStore.Guardian[] memory signingGuardians = new SecurityStore.Guardian[](signers.length);
+        Data.Guardian[] memory signingGuardians = new Data.Guardian[](signers.length);
```

Comment:
in the for-loop below, should we enforce `signers[i] > lastSigner` to deal with dupliciate signers?

Reply:
The assumption is that the signatures are already checked at this point (as we don't check them here at all), so duplicate checking is also already done. Because of that I think we don't need to re-check those things here.